### PR TITLE
fix: dynamic generic fee handler fee calcualtion

### PR DESCRIPTION
**Pull request by nmlinaric**
## Description
Current implementation for calculating dynamic fee for permissionless generic handler deposits won't work since it relies on `_resourceIDToTokenContractAddress` mapping existing in `PermissionlessGenericHadnerl`, this introduces a check to differentiate dynamic fee calculation for ERC20 generic deposits.

## Related Issue Or Context
Closes: #137 

## How Has This Been Tested? Testing details.
unit tests, e2e tests

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation

## Checklist:
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [ ] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: test/handlers/fee/dynamic/collectFeeGenericEVM.js [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+      msgGasLimit,
+    };
+
+    const feeData = Helpers.createOracleFeeData(
+      oracleResponse,
+      oracle.privateKey,
+      tokenAmount
+    );
+
+    const {fee} = await FeeHandlerRouterInstance.calculateFee.call(
```
Comment (lastperson) [inline test/handlers/fee/dynamic/collectFeeGenericEVM.js:139]: If the fee is known to be "0.000036777" then perhaps no need to call calculateFee() ?

File: test/handlers/fee/dynamic/collectFeeGenericEVM.js [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          value: Ethers.utils.parseEther("0.5").toString(),
+        }
+      )
+    );
```
Comment (lastperson) [inline test/handlers/fee/dynamic/collectFeeGenericEVM.js:254]:
```suggestion
    // Covered by test: deposit should revert if invalid fee (msg.value) amount supplied
```

File: test/handlers/fee/dynamic/collectFeeGenericEVM.js [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+      DynamicGenericFeeHandlerEVMInstance.collectFee(
+        depositorAddress,
+        originDomainID,
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          value: Ethers.utils.parseEther("0.5").toString(),
```
Comment (lastperson) [inline test/handlers/fee/dynamic/collectFeeGenericEVM.js:285]: All test values should be valid, except for the one that is being tested.
```suggestion
          value: Ethers.utils.parseEther("0.000036777"),
```

File: test/handlers/fee/dynamic/collectFeeGenericEVM.js [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+      FeeHandlerRouterInstance.collectFee(
+        depositorAddress,
+        originDomainID,
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          value: Ethers.utils.parseEther("0.5").toString(),
```
Comment (lastperson) [inline test/handlers/fee/dynamic/collectFeeGenericEVM.js:320]:
```suggestion
          value: Ethers.utils.parseEther("0.000036777"),
```

File: test/handlers/fee/dynamic/collectFeeGenericEVM.js [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+      msgGasLimit,
+    };
+
+    const feeData = Helpers.createOracleFeeData(
+      oracleResponse,
+      oracle.privateKey,
+      tokenAmount
+    );
+
+    const {fee} = await FeeHandlerRouterInstance.calculateFee.call(
```
Comment (lastperson) [inline test/handlers/fee/dynamic/collectFeeGenericEVM.js:349]: Fee is already known, no need to call calculateFee.

**Review by lastperson — CHANGES_REQUESTED** [review]

**Review by mpetrun5 — APPROVED** [review]

File: test/handlers/fee/dynamic/collectFeeGenericEVM.js [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+      DynamicGenericFeeHandlerEVMInstance.collectFee(
+        depositorAddress,
+        originDomainID,
+        destinationDomainID,
+        resourceID,
+        depositData,
+        feeData,
+        {
+          from: depositorAddress,
+          fee,
```
Comment (lastperson) [inline test/handlers/fee/dynamic/collectFeeGenericEVM.js:211]:
```suggestion
          value: fee,
```

**Review by lastperson — APPROVED** [review]

**Pull request merged**